app/workers/concerns/dump_emails.rb

```diff
@@ -0,0 +1,28 @@
+module DumpEmails
+  def emails_to_csv_file
+    CSV.open(csv_file_path, 'wb') do |csv|
+      user_emails.find_in_batches do |user_batch|
+        user_batch.each { |user| csv << [ user.email ] }
+      end
+    end
+  end
+
+  def write_to_s3
+    storage_path = MediaStorage.stored_path("application/x-gzip", "email_exports")
+    prefix = File.dirname(storage_path)
+    file_paths = File.basename(storage_path).split(".")
+    file_paths.shift
+    exts = file_paths.join(".")
+    file_name = "#{export_type}_email_list"
+    s3_path = "#{prefix}/#{file_name}.#{exts}"
+    MediaStorage.put_file(s3_path, gzip_file_path, storage_opts(file_name))
```

Comment:
This may need to go into another bucket path (not static) here. If so we'd need to initialize the AwsAdapter with the specific bucket path for the put operations.